From b5348dea8dad6a70c7fb7ef0ee490c599dde0f14 Mon Sep 17 00:00:00 2001
From: Moria Zuberi <moria.zuberi@tandemg.com>
Date: Mon, 12 Jul 2021 12:22:44 +0300
Subject: [PATCH] [SW-48525] gaudi2: ppboot: expose pcie init func

Move ppboot_pcie_init() to be publicly declared in pci.h
instead of staticaly defined for ppboot and change name to
hl_pcie_init().
This is done since eROM code will also need to use the same
functionality.

Change-Id: Iedde9556b4ddbbdd8c11f412a88b796f5618092b
Signed-off-by: Moria Zuberi <moria.zuberi@tandemg.com>
---

diff --git a/app/ppboot/ppboot.c b/app/ppboot/ppboot.c
index 3edaf07..7b4051f 100644
--- a/app/ppboot/ppboot.c
+++ b/app/ppboot/ppboot.c
@@ -20,6 +20,7 @@
 #include "boot_desc.h"
 #include "pll.h"
 #include <boot_if.h>
+#include <pci.h>
 
 __weak bool ppboot_is_valid_img_desc(struct boot_desc *desc)
 {
@@ -180,7 +181,7 @@
 
 __weak void ppboot_log_waiting_for_pcie(void) {}
 
-static void ppboot_pcie_init(void)
+void hl_pcie_init(void)
 {
 	if (hl_pcie_link_up()) {
 		hl_pcie_enable_access();
@@ -252,7 +253,7 @@
 	preboot_sync_btl();
 	hl_uart_init();
 	spi_prot_early();
-	ppboot_pcie_init();
+	hl_pcie_init();
 	ppboot_pcie_drain_init();
 	ppboot_version();
 	rst_src_dump();
diff --git a/lib/common/pci.h b/lib/common/pci.h
index 2d03dc5..4a1665a 100644
--- a/lib/common/pci.h
+++ b/lib/common/pci.h
@@ -63,4 +63,6 @@
 u32 dev_pci_dbi_cfg_offset_get(void);
 u32 hl_pci_get_ven_id_dev_id(void);
 
+void hl_pcie_init(void);
+
 #endif /* _PCI_H_ */
